### ISSUE TITLE
krun error on Windows 10

### ISSUE THREAD

**Issue by grosu** [kframework/k-legacy#1992]
I'm trying to run the lessons in 1_lambda and I'm getting the following error when running krun:

```
D:\github\grosu\k\k-distribution\tutorial\1_k\1_lambda\lesson_1>kompile lambda.k

[Warning] Compiler: Could not find main syntax module with name LAMBDA-SYNTAX
in definition.  Use --syntax-module to specify one. Using LAMBDA as default.
D:\github\grosu\k\k-distribution\tutorial\1_k\1_lambda\lesson_1>krun tests\ident
ity.lambda
[Error] Critical: Error while running process:Cannot run program "kast" (in
directory "."): CreateProcess error=2, The system cannot find the file
specified
D:\github\grosu\k\k-distribution\tutorial\1_k\1_lambda\lesson_1>
```

Note that I have not run K for a long time, but I would expect the above to work.

**Comment by grosu** [comment]
@laurayuwen , @daejunpark , did you guys have problems with this?

**Comment by grosu** [comment]
This is what I get with --debug:

```
D:\github\grosu\k\k-distribution\tutorial\1_k\1_lambda\lesson_1>krun --debug tes
ts\identity.lambda
org.kframework.utils.errorsystem.KEMException: [Error] Critical: Error while run
ning process:Cannot run program "kast" (in directory "."): CreateProcess error=2
, The system cannot find the file specified
        at org.kframework.utils.errorsystem.KEMException.create(KEMException.jav
a:101)
        at org.kframework.utils.errorsystem.KEMException.criticalError(KEMExcept
ion.java:34)
        at org.kframework.krun.RunProcess.execute(RunProcess.java:73)
        at org.kframework.krun.KRun.externalParse(KRun.java:315)
        at org.kframework.krun.KRun.parseConfigVars(KRun.java:263)
        at org.kframework.krun.KRun.run(KRun.java:70)
        at org.kframework.krun.KRunFrontEnd.run(KRunFrontEnd.java:91)
        at org.kframework.main.FrontEnd.main(FrontEnd.java:52)
        at org.kframework.main.Main.runApplication(Main.java:110)
        at org.kframework.main.Main.runApplication(Main.java:100)
        at org.kframework.main.Main.main(Main.java:52)
Caused by: java.io.IOException: Cannot run program "kast" (in directory "."): Cr
eateProcess error=2, The system cannot find the file specified
        at java.lang.ProcessBuilder.start(ProcessBuilder.java:1048)
        at org.kframework.krun.RunProcess.execute(RunProcess.java:43)
        ... 8 more
Caused by: java.io.IOException: CreateProcess error=2, The system cannot find th
e file specified
        at java.lang.ProcessImpl.create(Native Method)
        at java.lang.ProcessImpl.<init>(ProcessImpl.java:386)
        at java.lang.ProcessImpl.start(ProcessImpl.java:137)
        at java.lang.ProcessBuilder.start(ProcessBuilder.java:1029)
        ... 9 more
java.io.IOException: Cannot run program "kast" (in directory "."): CreateProcess
 error=2, The system cannot find the file specified
        at java.lang.ProcessBuilder.start(ProcessBuilder.java:1048)
        at org.kframework.krun.RunProcess.execute(RunProcess.java:43)
        at org.kframework.krun.KRun.externalParse(KRun.java:315)
        at org.kframework.krun.KRun.parseConfigVars(KRun.java:263)
        at org.kframework.krun.KRun.run(KRun.java:70)
        at org.kframework.krun.KRunFrontEnd.run(KRunFrontEnd.java:91)
        at org.kframework.main.FrontEnd.main(FrontEnd.java:52)
        at org.kframework.main.Main.runApplication(Main.java:110)
        at org.kframework.main.Main.runApplication(Main.java:100)
        at org.kframework.main.Main.main(Main.java:52)
Caused by: java.io.IOException: CreateProcess error=2, The system cannot find th
e file specified
        at java.lang.ProcessImpl.create(Native Method)
        at java.lang.ProcessImpl.<init>(ProcessImpl.java:386)
        at java.lang.ProcessImpl.start(ProcessImpl.java:137)
        at java.lang.ProcessBuilder.start(ProcessBuilder.java:1029)
        ... 9 more
[Error] Critical: Error while running process:Cannot run program "kast" (in
directory "."): CreateProcess error=2, The system cannot find the file
specified
```

**Comment by dwightguth** [comment]
Pretty sure this is because we are now using external parsers for everything and you are not running from a cygwin environment so it doesn't know how to interpret and execute the bash script. The simple fix should be to add a .bat to the default on windows. I will take care of this.

**Comment by grosu** [comment]
And, of course, kast works in command line, otherwise I would have reported a different bug:

D:\github\grosu\k\k-distribution\tutorial\1_k\1_lambda\lesson_1>kast tests\identity.lambda
`lambda_._`(#token("x","Id"),#token("x","Id"))

**Comment by grosu** [comment]
Thanks, @dwightguth!  I think we are already doing that for all our RV tools, so that should work.

**Comment by radumereuta** [comment]
@daejunpark but that means it's not limited only to windows 10, but to all windows platforms.
I know that Java has this issue where if you try to execute a .bat file, you have to specify the extension. And it's not required in the command line.

Another thing. Why aren't you calling the parser directly from krun? It's more efficient, and you avoid all kinds of problems.

**Comment by dwightguth** [comment]
@radumereuta the old implementation was extremely quirky as a result of what we did. I'm not at all convinced that the performance difference associated with a new process is particularly noticeable with nailgun. If you want to try to test it and compare and let me know if you still see a performance difference, let me know. But I suspect it will be about the same either way.

I am choosing to go a different route this time by making the default be an absolute path to the  script on disk so that we avoid all the problems with path issues that we had before while still maintaining a much cleaner implementation.

**Comment by cos** [comment]
@dwightguth, regarding the second issue, you may want to merge my other PR then.

**Comment by cos** [comment]
Regarding the first, I agree with @radumereuta and the two of us had this discussion before. You may like the command line as the "API" of the tool, but it isn't the way it's usually done and can lead to problems like this.

**Comment by radumereuta** [comment]
I actually agree with @dwightguth on this one.
The only problem I can foresee is having a good enough intermediate language to represent the terms and error messages. In this case is KORE, and we are gaining confidence in the implementation every day.

**Comment by cos** [comment]
@radumereuta, the issue I'm referring to is having the command line as the API of K. This is a bad idea, and I'm not even going to argue about it anymore.

**Comment by dwightguth** [comment]
If you can come up with a way to have krun call an API for kast in a way that allows support for external parsers and is cleaner than the way it was implemented in the old implementation, I have no particular objection. It's just not remotely a priority for me to work on since we don't use kast at all in the C semantics.

**Comment by grosu** [comment]
We do not have time for this now, please let's worry about it after Jan 18.  The simple bug fix is good enough for now.

**Comment by cos** [comment]
fixed by #1994
